### ISSUE TITLE
json-ld vs xml-ld

### ISSUE THREAD

**Issue by Friedrich777** [json-ld/json-ld.org#752]
Dear developers,
please develop XML-LD, then the world of web technologies will come to order. You are well aware that XSD technology solves all problems to avoid mistakes. And as a developer, it will be convenient for me to create an XSD schema and be happy.
Thanks in advance
Friedrich

**Comment by Friedrich777** [comment]
`<ld:xml xmlns:ld="https://xml-ld.org/">
  <ld:context>http://schema.org</ld:context>
  <ld:type>NewsArticle</ld:type>
  <mainEntityOfPage>
    <ld:type>WebPage</ld:type>
    <ld:id>https://viziya.info/tochka-zreniya/politika/protesty-ugrozhayut-obshchestvu-uvereny-v-partii-sluga-naroda.html</ld:id>
  </mainEntityOfPage>
  <headline>Протесты угрожают обществу, уверены в партии «Слуга Народа»</headline>
  <alternativeHeadline>Протесты под ОП могут обернуться войной между СБУ и МВД</alternativeHeadline>
  <image>
    <ld:type>ImageObject</ld:type>
    <url>https://viziya.info/pic/2021/03/1599.jpg</url>
  </image>
  <datePublished>2021-03-22T17:58:34+02:00</datePublished>
  <dateModified>2021-03-22T17:58:34+02:00</dateModified>
  <author>
    <ld:type>Person</ld:type>
    <name>Всеволод Гордиенко</name>
  </author>
  <publisher>
    <ld:type>Organization</ld:type>
    <name>Визия</name>
    <brand>ВизиЯ</brand>
    <alternateName>Viziya</alternateName>
    <logo>
      <ld:type>ImageObject</ld:type>
      <url>https://viziya.info/img/slogo.png</url>
      <width>212</width>
      <height>60</height>
    </logo>
  </publisher>
</ld:xml>`

**Comment by pchampin** [comment]
If you want to serialize Linked Data in XML, why not simply use [RDF/XML](https://www.w3.org/TR/rdf-syntax-grammar/) ?

**Comment by gkellogg** [comment]
> RDF is also becoming even more complex with RDF* which does not currently map to JSON-LD

FYI, see json-ld.github.io/json-ld-star.

**Comment by bertvannuffelen** [comment]
Even it is closed I like to add some notes for the XML community:

RDF/XML is a serialization of the RDF graph into XML. For that reason there is no fixed business-oriented XSD possible. It is just a technical sound serialization. And for that reason the only valid advice to XML developers is not to use RDF/XML as the basis for development. Your basic assumptions on stability/verification/etc. will not work. 
I would not call it a "failed experiment". But the naive expectation that this is the bridge between the XML community and the Semantic Web community will be very disappointing. So better don't try it. (Even more: there are fundamental differences between qNames in XML and RDF).

On the request for XML-ld, that is something for which I hear a lot of support in governments. They use XML data exchanges which should align with the semantics, expressed as a semantic model.  There is no need for them to switch to json-serialization data exchanges (a change just for spending money is not very encouraged ;-). But there is a need to ensure that the XML content aligns with the semantic model.
For that XML-ld mapping approach would be great. Today everybody has to develop their own mapping approach. Standardizing this will bridge it.

But I agree this is not the objective of this working group. It should be initiated by the XML community. Suggestion, maybe this should be requested at https://www.w3.org/XML/EXI/.

**Comment by pchampin** [comment]
Another effort is this direction was [GRDDL](https://www.w3.org/TR/grddl-primer/) but I am not sure how much it is still used (and implemented) nowadays.

**Comment by gkellogg** [comment]
As JSON-LD algorithms are defined in terms of an [internal representation](https://www.w3.org/TR/json-ld11-api/#dfn-internal-representation) many different serialization formats could be used (YAML and CBOR described as examples). Conceivably, XML could as well, although the use of namespaces in XML, and term definitions used as namespaces in JSON-LD contexts overlap, and could naively lead to incorrect XML. But, getting into XML brings up many issues.

Another place to look might be [RDFa](https://www.w3.org/TR/rdfa-core/). Part of the [test suite](http://rdfa.info/test-suite/) works over RDFa in XML. Although it's intended for HTML, it can work in other similar SGML-derived formats.

**Comment by niklasl** [comment]
A long time ago (when I regularly worked with XML alongside RDF), I drafted a replacement for RDF/XML, called [Grit](https://github.com/oort/oort/blob/wiki/Grit.md) ("Grokkable RDF Is Transformable"). That was on Google Code, but I exported that to Github when that closed down.  (the XSLT implementation is at: https://github.com/oort/oort/blob/master/impl/xslt/grit/rdfxml-grit.xslt and the GRDDL back to RDF/XML is at: https://github.com/oort/oort/blob/master/impl/xslt/grit/grit-grddl.xslt .)

(I would probably still recommend that kind of approach over attempting the JSON-LD design in XML without leveraging XML namespaces, although I can see why e.g. type coercion would be attractive. Also, the Grit design expected the default namespace to be the null namespace, so it would need an overhaul to support the nowadays common practise of default vocabularies.)
